.github/workflows/artifact-tests.yml

```diff
@@ -0,0 +1,144 @@
+name: artifact-unit-tests
+on:
+  push:
+    paths:
+        - 'packages/artifact/**'
+  pull_request:
+    paths:
+        - 'packages/artifact/**'
+
+jobs:
+
+  build:
+    name: Build
+
+    strategy:
+      matrix:
+        runs-on: [ubuntu-latest]
+      fail-fast: false
+
+    runs-on: ${{ matrix.runs-on }}
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+
+    - name: Set Node.js 12.x
+      uses: actions/setup-node@v1
+      with:
+        node-version: 12.x
+
+    # Need root node_modules because certain npm packages like jest are configured for the entire repository and it won't be possible
+    # without these to just compile the artifacts package
+    - name: Install root npm packages
+      run: npm install
+
+    - name: Compile artifact package
+      run: |
+        npm install
+        tsc
+      working-directory: packages/artifact
+
+    - name: Create files that will be uploaded
+      run: |
+        mkdir artifact-path
+        echo "hello" > artifact-path/world.txt
+        echo "Some large amount of text that has a compression ratio that is greater than 100%. If greater than 100%, gzip is used to upload the file" > artifact-path/gzip.txt
+
+    # Certain env variables are needed in order to make the necessary API calls. These env variables are not available in a standard shell but they
+    # are available for actions written in node. We're using a hack here to get the necessary env variables from a different context
+    - name: Get ACTIONS_RUNTIME_TOKEN env variable
+      uses: konradpabjan/Get-Actions-Env-Variable-From-Node-Context@4a3142ee3e
+      with:
+        EnvName: ACTIONS_RUNTIME_TOKEN
+      id: get_runtime_token
+
+    - name: Get ACTIONS_RUNTIME_URL env variable
+      uses: konradpabjan/Get-Actions-Env-Variable-From-Node-Context@4a3142ee3e
+      with:
+        EnvName: ACTIONS_RUNTIME_URL
+      id: get_runtime_url
+
+    - name: Get GITHUB_RUN_ID env variable
+      uses: konradpabjan/Get-Actions-Env-Variable-From-Node-Context@4a3142ee3e
+      with:
+        EnvName: GITHUB_RUN_ID
+      id: get_run_id
+
+    # The env variables that were retrieved in a different step need to be set for each subsequent shell or upload/download will fail
+    # We're using node -e to call the functions directly available in the @actions/artifact package
+    - name: Upload artifacts using uploadArtifact()
+      run: |
+        export ACTIONS_RUNTIME_TOKEN="${{ steps.get_runtime_token.outputs.EnvValue }}"
```

Comment:
`${{ github.token }}` does not work unfortunately so I'm having to use this

Reply:
I also tried permanently setting the env variables in a separate step for the machine so it's global but that didn't work so I'm having to repeat the same 3 steps multiple times

Reply:
You should be able to use https://help.github.com/en/actions/reference/workflow-commands-for-github-actions#setting-an-environment-variable to do this

Reply:
Forgot about this, it does the trick! I tried a bunch of things in bash and powershell to globally set it but it would never work for some reason.